bundles/org.openhab.binding.bluetooth/src/main/java/org/openhab/binding/bluetooth/discovery/internal/BluetoothDiscoveryProcess.java

```diff
@@ -63,14 +66,28 @@
     private final Lock serviceDiscoveryLock = new ReentrantLock();
     private final Condition connectionCondition = serviceDiscoveryLock.newCondition();
     private final Condition serviceDiscoveryCondition = serviceDiscoveryLock.newCondition();
-    private final Condition nameDiscoveryCondition = serviceDiscoveryLock.newCondition();
+    private final Condition infoDiscoveryCondition = serviceDiscoveryLock.newCondition();
 
     private final BluetoothDevice device;
     private final Collection<BluetoothDiscoveryParticipant> participants;
     private final Set<BluetoothAdapter> adapters;
 
     private volatile boolean servicesDiscovered = false;
 
+    /**
+     * List of UUID's which will be fetched from the device when connection based discovery is required.
+     */
+    private final List<UUID> deviceInformationUUIDs = new ArrayList<UUID>(Arrays.asList(
+            GattCharacteristic.DEVICE_NAME.getUUID(), GattCharacteristic.MODEL_NUMBER_STRING.getUUID(),
+            GattCharacteristic.SERIAL_NUMBER_STRING.getUUID(), GattCharacteristic.HARDWARE_REVISION_STRING.getUUID(),
+            GattCharacteristic.FIRMWARE_REVISION_STRING.getUUID(),
+            GattCharacteristic.SOFTWARE_REVISION_STRING.getUUID()));
```

Comment:
I don't see the benefit of putting this in a list if you have to do a switch statement on each of them anyway.
I think a better way would be to have a helper function that takes a characteristic as parameter instead and you conditionally call the helper function based on the presence or absence of the device field of interest.